internal/service/alarms/internal/db/repo/alarms_repository.go

```diff
@@ -0,0 +1,45 @@
+package repo
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/openshift-kni/oran-o2ims/internal/service/alarms/internal/db/models"
+)
+
+// All DB interaction code goes here
+
+type AlarmsRepository struct {
+	Db *pgxpool.Pool
+}
+
+// GetAlarmEventRecordWithUuid grabs a row of alarm_event_record using uuid
+func (ar *AlarmsRepository) GetAlarmEventRecordWithUuid(ctx context.Context, uuid uuid.UUID) ([]models.AlarmEventRecord, error) {
+	// Build sql query
+	query := `
+SELECT alarm_event_record_id, alarm_definition_id, probable_cause_id, alarm_raised_time, 
+       alarm_changed_time, alarm_cleared_time, alarm_acknowledged_time, 
+       alarm_acknowledged, perceived_severity, extensions, resource_type_id
+FROM alarm_event_record 
+WHERE alarm_event_record_id = $1
+`
+
+	// Run query
+	rows, _ := ar.Db.Query(ctx, query, uuid.String()) // note: err is passed on to Collect* func so we can ignore this
```

Comment:
I assume the answer is yes, but are we sure that the `Query` method protects us against SQL injection attacks? ... i know in this particular case it isn't possible since the UUID string would be validated for format before this point, but in other instances that may not always be the case.

Reply:
Yeah basically using `$` syntax should prevent any injection issues! The author talks about it [here](https://github.com/jackc/pgx/discussions/2099#discussioncomment-10231081) a bit more. There's also a helper function called [NamedArgs](https://pkg.go.dev/github.com/jackc/pgx/v5#NamedArgs) (not used here) is also safe.